sample-app/src/main/java/com/joinforage/android/example/ui/pos/POSViewModel.kt

```diff
@@ -129,4 +158,28 @@ class POSViewModel : ViewModel() {
             }
         }
     }
+
+    fun capturePayment(foragePinEditText: ForagePINEditText, terminalId: String, paymentRef: String, onSuccess: (response: PaymentResponse?) -> Unit) {
+        viewModelScope.launch {
+            val response = ForageTerminalSDK(terminalId).capturePayment(
+                CapturePaymentParams(
+                    foragePinEditText = foragePinEditText,
+                    paymentRef = paymentRef
+                )
+            )
+
+            when (response) {
+                is ForageApiResponse.Success -> {
+                    val moshi = Moshi.Builder().build()
+                    val jsonAdapter: JsonAdapter<PaymentResponse> = PaymentResponseJsonAdapter(moshi)
+                    val paymentResponse = jsonAdapter.fromJson(response.data)
+                    _uiState.update { it.copy(paymentResponse = paymentResponse) }
+                    onSuccess(paymentResponse)
+                }
+                is ForageApiResponse.Failure -> {
+                    Log.e("POSViewModel", response.toString())
+                }
```

Comment:
I guess it took creating a Sample App for me to realize it but it's lame that we don't return typed responses and that clients are burdened to parse the JSON string response. Definitely an update we should release at some point